### PR TITLE
BACKLOG-12299: Upgrade react-apollo and add alias for replacing compose

### PR DESCRIPTION
**Pull request by FBerthelot**
## JIRA

https://jira.jahia.org/browse/BACKLOG-12299

## Tests

The following are included in this PR
- [ ] Unit Tests (Most changes _should_ have unit tests)
- [ ] Integration Tests

## Checklist

I have considered the following implications of my change: 

- [ ] Security (in particular for changes to authentication, authorization, data fetching, ...)
- [ ] Performance
- [ ] Migration
- [ ] Code maintainability

## Documentation

- [ ] Inline documentation
- [ ] Internal Documentation (wiki)
- [ ] User-facing Documentation

### CODE REVIEW

File: src/javascript/JContent/ContentRoute/ContentLayout/PreviewDrawer/Preview/Preview.jsx [hunk 1]
```diff
@@ -63,51 +63,58 @@ const styles = theme => ({
...
+    useEffect(() => {
+        if (!loading && !error) {
+            refetch();
         }
+    });
 
-        if (_.isEmpty(previewSelection)) {
-            return <NoPreviewComponent {...this.props}/>;
-        }
+    if (selection.length > 0) {
```
Comment (Taybou) [inline src/javascript/JContent/ContentRoute/ContentLayout/PreviewDrawer/Preview/Preview.jsx]: WHAT! :D

File: src/javascript/JContent/ContentRoute/ContentLayout/PreviewDrawer/Preview/Preview.jsx [hunk 2]
```diff
@@ -63,51 +63,58 @@ const styles = theme => ({
...
+        return <ProgressOverlay/>;
+    }
+
+    return (
+        <PreviewComponent data={data.jcr ? data.jcr : {}}
+                          workspace={previewMode}
+                          fullScreen={(previewState === CM_DRAWER_STATES.FULL_SCREEN)}
+        />
+    );
+};
```
Comment (Taybou) [inline src/javascript/JContent/ContentRoute/ContentLayout/PreviewDrawer/Preview/Preview.jsx]: WHAT WHAT :D

**Comment by DameniMilo** [comment]
Seems like you have some conflict to resolve first :p

File: jest.config.js [hunk 3]
```diff
@@ -1,3 +1,9 @@
 const jestConfig = require('@jahia/test-framework').jestConfig;
 
-module.exports = jestConfig;
+module.exports = {
+    ...jestConfig,
+    moduleNameMapper: {
+        ...jestConfig.moduleNameMapper,
+        '~/(.*)': '<rootDir>/src/javascript/$1'
```
Comment (DameniMilo) [inline jest.config.js]: If I'm not mistaking you have the same exact line in content-editor, so instead of adding this here could you add it to the shared config and just update the test-framework version?

File: src/javascript/JContent/ContentRoute/ContentLayout/ContentLayout.container.jsx [hunk 4]
```diff
@@ -306,7 +307,7 @@ ContentLayoutContainer.propTypes = {
     sort: PropTypes.object.isRequired,
     t: PropTypes.func.isRequired,
     uilang: PropTypes.string.isRequired,
-    treeState: PropTypes.number.isRequired,
+    treeState: PropTypes.number,
```
Comment (DameniMilo) [inline src/javascript/JContent/ContentRoute/ContentLayout/ContentLayout.container.jsx]: Why did you remove the isRequired?

Comment (FBerthelot) [reply src/javascript/JContent/ContentRoute/ContentLayout/ContentLayout.container.jsx]: proptype error in console that anoyed me. As the component work well without this props, I removed the required here.

File: jest.config.js (hunk 3 above)
Comment (FBerthelot) [reply jest.config.js]: You can add it to the test-framework version :)

File: src/javascript/JContent/ImageEditor/ImageEditor.container.test.jsx [hunk 5]
```diff
@@ -90,7 +88,7 @@ describe('Image Edition', () => {
         wrapper = mount(
             <DxContext.Provider value={dxContext}>
                 <MuiThemeProvider theme={theme}>
-                    <MockedProvider mocks={mocks} cache={new InMemoryCache({fragmentMatcher})}>
+                    <MockedProvider mocks={mocks}>
```
Comment (DameniMilo) [inline src/javascript/JContent/ImageEditor/ImageEditor.container.test.jsx]: why do you remove the cache here?

File: src/javascript/utils/compose.js [hunk 6]
```diff
@@ -0,0 +1 @@
+export const compose = (...funcs) => funcs.reduce((a, b) => (...args) => a(b(...args)), arg => arg);
```
Comment (DameniMilo) [inline src/javascript/utils/compose.js]: are we not using this in other projects too? so should we not put this code somewhere we can share it?

File: jest.config.js (hunk 3 above)
Comment (DameniMilo) [reply jest.config.js]: Since you are introducing it, you will have to do so :)

File: src/javascript/JContent/ContentRoute/ContentLayout/ContentLayout.container.jsx (hunk 4 above)
Comment (DameniMilo) [reply src/javascript/JContent/ContentRoute/ContentLayout/ContentLayout.container.jsx]: Ok could you add a default value then :)

Comment (FBerthelot) [reply src/javascript/JContent/ContentRoute/ContentLayout/ContentLayout.container.jsx]: What kind of default value ?

File: src/javascript/JContent/ImageEditor/ImageEditor.container.test.jsx (hunk 5 above)
Comment (FBerthelot) [reply src/javascript/JContent/ImageEditor/ImageEditor.container.test.jsx]: No need for cache in a test 🤕

File: src/javascript/utils/compose.js (hunk 6 above)
Comment (FBerthelot) [reply src/javascript/utils/compose.js]: Were you want to put it ? data-helper is not an option, as we already discuss.

I think we can keep this here, as we will progressively delete HOC pattern for use hooks.

File: jest.config.js (hunk 3 above)
Comment (FBerthelot) [reply jest.config.js]: Bye bye, you will fix this, I'm week off. Ciao :p

File: src/javascript/JContent/ImageEditor/ImageEditor.container.test.jsx (hunk 5 above)
Comment (DameniMilo) [reply src/javascript/JContent/ImageEditor/ImageEditor.container.test.jsx]: ah yep didn't see it was a test ^^

File: src/javascript/JContent/ContentRoute/ContentLayout/ContentLayout.container.jsx (hunk 4 above)
Comment (DameniMilo) [reply src/javascript/JContent/ContentRoute/ContentLayout/ContentLayout.container.jsx]: The property has been removed actually
